### PR TITLE
Add  de.uni_heidelberg.zah.GaiaSky

### PR DESCRIPTION
**Pull request by BryanQuigley**
Upstream discussion: https://gitlab.com/langurmonkey/gaiasky/-/issues/337

I do get a warning about the icon not being present, but it seems to work fine.

My first flatpak, so be critical - I want to learn best practices. Thanks!

### CODE REVIEW

File: de.uni_heidelberg.zah.GaiaSky.desktop [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Type=Application
+Encoding=UTF-8
+Name=Gaia Sky
+Comment=Gaia Sky 3D Universe software
+Exec=gaiasky
+Icon=de.uni_heidelberg.zah.GaiaSky.svg
```
Comment (bilelmoussaoui) [inline de.uni_heidelberg.zah.GaiaSky.desktop:7]:
```suggestion
Icon=de.uni_heidelberg.zah.GaiaSky
```

File: de.uni_heidelberg.zah.GaiaSky.desktop [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+Terminal=false
```
Comment (bilelmoussaoui) [inline de.uni_heidelberg.zah.GaiaSky.desktop:8]: Nitpick: You can add `Keywords` as well.

File: de.uni_heidelberg.zah.GaiaSky.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  - --env=JAVA_HOME=/app/jre
+modules:
+  - name: openjdk
+    buildsystem: simple
+    build-commands: [/usr/lib/sdk/openjdk11/install.sh]
+
+  - name: gaiasky
+    buildsystem: simple
+    build-commands:
+       - mkdir -p /app/bin/bookmarks/ /app/bin/conf/ /app/bin/i18n/ /app/bin/lib/ /app/bin/mappings/ /app/bin/music/ /app/bin/scripts/
```
Comment (bilelmoussaoui) [inline de.uni_heidelberg.zah.GaiaSky.yaml:23]: Nitpick: Any reason for not using install here as well?

File: de.uni_heidelberg.zah.GaiaSky.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+       - cp -R scripts/* /app/bin/scripts/
+       - rm /app/bin/lib/*-macos.jar
+       - rm /app/bin/lib/*-windows.jar
+       - install -D gs_icon.svg /app/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg
+       - install -D gaiasky /app/bin/gaiasky
+       - install -D de.uni_heidelberg.zah.GaiaSky.desktop /app/share/applications/${FLATPAK_ID}.desktop
+       - install -D de.uni_heidelberg.zah.GaiaSky.xml /app/share/metainfo/${FLATPAK_ID}.xml
+    sources:
+      - type: archive
+        url: https://gaia.ari.uni-heidelberg.de/gaiasky/files/releases/2.3.1.b19516f2e/gaiasky-2.3.1.b19516f2e.tar.gz
```
Comment (bilelmoussaoui) [inline de.uni_heidelberg.zah.GaiaSky.yaml:39]: You could setup [flathub-external-data-checker](https://github.com/flathub/flatpak-external-data-checker#flatpak-external-data-checker) to auto-update the app.

File: de.uni_heidelberg.zah.GaiaSky.xml [hunk 5]
```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bilelmoussaoui) [inline de.uni_heidelberg.zah.GaiaSky.xml:1]: nitpick: Not sure why the original filename doesn't use `.metainfo.xml`

File: de.uni_heidelberg.zah.GaiaSky.xml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    <control>keyboard</control>
+    <control>gamepad</control>
+  </recommends>
+  
+  <description>
+    <p>
+      Gaia Sky is a real-time, 3D, astronomy visualisation software for desktop and VR that runs on Windows, Linux and macOS. It is developed in the framework of ESA&apos;s Gaia mission to chart about 1 billion stars of our Galaxy in the Gaia group of the Astronomisches Rechen-Institut (ZAH, Universität Heidelberg). Explore the Solar System, the Milky Way or even the most distant galaxies of the Universe.
+    </p>
+  </description>
+
```
Comment (bilelmoussaoui) [inline de.uni_heidelberg.zah.GaiaSky.xml:22]: You need a releases tag & a OARS one as well.

**Comment by bilelmoussaoui** [comment]
bot, build de.uni_heidelberg.zah.GaiaSky

File: de.uni_heidelberg.zah.GaiaSky.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+       - cp -R lib/* /app/bin/lib/
+       - cp -R mappings/* /app/bin/mappings/
+       - cp -R music/* /app/bin/music/
+       - cp -R scripts/* /app/bin/scripts/
+       - rm /app/bin/lib/*-macos.jar
+       - rm /app/bin/lib/*-windows.jar
+       - install -D gs_icon.svg /app/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg
+       - install -D gaiasky /app/bin/gaiasky
+       - install -D de.uni_heidelberg.zah.GaiaSky.desktop /app/share/applications/${FLATPAK_ID}.desktop
+       - install -D de.uni_heidelberg.zah.GaiaSky.xml /app/share/metainfo/${FLATPAK_ID}.xml
```
Comment (bilelmoussaoui) [inline de.uni_heidelberg.zah.GaiaSky.yaml:36]:
```suggestion
       - install -D de.uni_heidelberg.zah.GaiaSky.metainfo.xml /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
```
the build fails due to this.

File: de.uni_heidelberg.zah.GaiaSky.yaml (hunk 3 above)
Comment (BryanQuigley) [reply de.uni_heidelberg.zah.GaiaSky.yaml:23]: By my reading install can't copy entire directories.. maybe I got it wrong?

File: de.uni_heidelberg.zah.GaiaSky.desktop (hunk 1 above)
Comment (BryanQuigley) [reply de.uni_heidelberg.zah.GaiaSky.desktop:7]: will do

File: de.uni_heidelberg.zah.GaiaSky.yaml (hunk 3 above)
Comment (bilelmoussaoui) [reply de.uni_heidelberg.zah.GaiaSky.yaml:23]: Never mind

File: de.uni_heidelberg.zah.GaiaSky.yaml (hunk 4 above)
Comment (BryanQuigley) [reply de.uni_heidelberg.zah.GaiaSky.yaml:39]: Made all other suggested changes (and found 1 other bug) except for this one.  Can auto-updating this be added later?

Comment (bilelmoussaoui) [reply de.uni_heidelberg.zah.GaiaSky.yaml:39]: Sure thing. It's a suggestion

File: de.uni_heidelberg.zah.GaiaSky.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
+app-id: de.uni_heidelberg.zah.GaiaSky
+runtime: org.freedesktop.Platform
+runtime-version: '19.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions: [org.freedesktop.Sdk.Extension.openjdk11]
+command: gaiasky
+finish-args:
+  - --socket=x11
+  - --socket=wayland
```
Comment (bilelmoussaoui) [inline de.uni_heidelberg.zah.GaiaSky.yaml:9]: Does it actually work with wayland?

Comment (BryanQuigley) [reply de.uni_heidelberg.zah.GaiaSky.yaml:9]: My guess was yes, but I don't know.

**Comment by bilelmoussaoui** [comment]
bot, build de.uni_heidelberg.zah.GaiaSky

**Comment by BryanQuigley** [comment]
working on properly validationg metadata file.. fixed urls.

**Comment by nedrichards** [comment]
bot, build de.uni_heidelberg.zah.GaiaSky

@BryanQuigley you can write a comment with that command too when you push an update to generate a new build.

File: de.uni_heidelberg.zah.GaiaSky.yaml (hunk 8 above)
Comment (nedrichards) [reply de.uni_heidelberg.zah.GaiaSky.yaml:9]: This needs to be tested for sure.

**Comment by nedrichards** [comment]
Other than the wayland testing lgtm

**Comment by BryanQuigley** [comment]
It works on a Wayland enabled gnome shell- but my Xeyes testing (https://medium.com/@bugaevc/how-to-easily-determine-if-an-app-runs-on-xwayland-or-on-wayland-natively-8191b506ab9a) indicates it's actually using Xwayland.

Given that, I should drop the wayland perm?

**Comment by BryanQuigley** [comment]
bot, build de.uni_heidelberg.zah.GaiaSky

Tested and confirmed that dropping wayland makes sense for now

**Comment by BryanQuigley** [comment]
Confirmed build works :)

File: de.uni_heidelberg.zah.GaiaSky.yaml [hunk 9]
```diff
@@ -0,0 +1,43 @@
+app-id: de.uni_heidelberg.zah.GaiaSky
+runtime: org.freedesktop.Platform
+runtime-version: '19.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions: [org.freedesktop.Sdk.Extension.openjdk11]
+command: gaiasky
+finish-args:
+  - --socket=x11
```
Comment (nedrichards) [inline de.uni_heidelberg.zah.GaiaSky.yaml:8]: Sorry, just noticed in reviewing again but you need to add `--share=ipc` here as well to get optimal performance on X11 per https://docs.flatpak.org/en/latest/sandbox-permissions.html

**Comment by BryanQuigley** [comment]
bot, build de.uni_heidelberg.zah.GaiaSky

**Comment by bilelmoussaoui** [comment]
Looks good, let us know if upstream is interested in having write access to the repository.

/merge

**Comment by BryanQuigley** [comment]
Thanks!

> Looks good, let us know if upstream is interested in having write access to the repository.
Please do add upstream to the repo too: https://github.com/langurmonkey/

**Pull request closed without merging**